File: lib/dog/runtime/signal.rb
#
# Copyright 2011 by Salman Ahmad (salman@salmanahmad.com).
# All rights reserved.
#
# Permission is granted for use, copying, modification, distribution,
# and distribution of modified versions of this work as long as the
# above copyright notice is included.
#

module Dog
  class Signal
    attr_accessor :call_track
    attr_accessor :schedule_track
    attr_accessor :pause
    attr_accessor :stop
    attr_accessor :exit
  end
end

File: src/main/java/dog/lang/Signal.java
/*
 *
 *  Copyright 2012 by Salman Ahmad (salman@salmanahmad.com).
 *  All rights reserved.
 *
 *  Permission is granted for use, copying, modification, distribution,
 *  and distribution of modified versions of this work as long as the
 *  above copyright notice is included.
 *
 */

package dog.lang;

public class Signal {
	public enum Type {
		RETURN,
		INVOKE,
		SCHEDULE,
		PAUSE,
		STOP,
		EXIT
	}

	public Signal() {

	}

	public Signal(Type type) {
		this.type = type;
	}

	public Signal(Type type, StackFrame frame) {
		this.type = type;
		this.stackFrame = frame;
	}

	public Type type = Type.RETURN;
	public StackFrame stackFrame = null;
}